New blog post: "install keras environment from scratch"

FabienLavocat opened the pull request:
New blog post on Installing Keras on a new OS, from installing Ubuntu to PyCharm

Commit:
Setup Keras post - Install Ubuntu

Commit:
Setup Environment: Install Keras

Commit:
Setup Keras - Install PyCharm

Commit:
Minor text changes in Install Keras Environment from scratch

Commit:
Add an example in the Setup Keras env.

FabienLavocat committed:
Minor change in text

fchollet commented on content/install-keras-environment-from-scratch.md:
```diff
@@ -0,0 +1,174 @@
+Title: Install a Keras environment from scratch
+Date: 2016-03-09
+Category: News
```
This should be "tutorials".

FabienLavocat replied:
I've changed the category

fchollet commented on content/install-keras-environment-from-scratch.md:
````diff
@@ -0,0 +1,174 @@
+sudo pip install keras
+```
+
+Here you are! You've installed the minimal packages to run your first script using Keras. To make sure everything works, keep your terminal open and enter the following commands:
````
This does not cover running things on GPU, which is fundamental to getting things done in deep learning (and is the part of the installation that people usually need help with). We would need to cover:
- installing the Nvidia Cuda drivers
- installing cuDNN
- installing TensorFlow, GPU version
- making sure that Keras is running on GPU (with cuDNN enabled) for both TensorFlow and Theano

Ideally we should make two sections: one about setting up Keras with the Theano backend, one about setting up Keras with the TensorFlow backend.

fchollet commented on content/install-keras-environment-from-scratch.md:
```diff
@@ -0,0 +1,174 @@
+# Install git to get the latest version of the next package
+sudo apt-get install git
+
+# Install Theano (backend for Keras)
```
I think we should cover the installation of TensorFlow as well (including the GPU version).

FabienLavocat replied:
I was thinking about doing a different post for that. This one is about setting up a very first deep learning environment for Deep Learning with Keras. This is the very minimal stuff you need to do/install to make it work.

fchollet commented on content/install-keras-environment-from-scratch.md:
````diff
@@ -0,0 +1,174 @@
+
+First, you must open a **Terminal** and run the script bellow, line by line to install all the requirements of Keras and then Keras itself.
+
+```python
````
This is a command line, the language should be specified as `sh`. Also applies to the code below.

fchollet commented on content/install-keras-environment-from-scratch.md:
```diff
@@ -0,0 +1,174 @@
+Title: Install a Keras environment from scratch
```
I would call it "Set up a Keras Deep Learning environment from scratch". At least "set up": you don't install an environment.

FabienLavocat replied:
I've changed the title

fchollet commented:
Looks great, but I think we need basically the following two sections (each with 2 sub-sections):
- setting up Keras on a workstation with a GPU
  - install Cuda
  - install cuDNN (v3)
  - Theano version:
    - GPU-specific Theano installation instructions 
    - make sure Keras is running on Theano with cuDNN/on GPU
  - TensorFlow version:
    - install the GPU-specific TF version (which is harder to install)
    - make sure Keras is running with cuDNN/on GPU
- setting up Keras to run only on CPU:
  - Theano version (this one bullet point is really what the tutorial is currently about)
  - TF version

FabienLavocat committed:
Category should be 'Tutorials'

FabienLavocat committed:
Change the title to 'Set up a Keras Deep Learning environment from scratch'

FabienLavocat commented:
I 100% agree with you @fchollet about the two new sections. As I said in a previous comment. I think we should keep this article as the very minimal to start playing with Keras.
A second article to let people install TensorFlow and use the GPU will be required, but in a second article (IMO)

FabienLavocat committed:
Change the name of the blog post

WaBlueKey committed:
Content-edit of final draft

Complete content edit for content, structure, and readability as well as wording, phrasing, spelling, punctuation, and grammar.

FabienLavocat committed:
Merge pull request #1 from WaBlueKey/content

Content-edit of final draft

FabienLavocat committed:
Change the name of the blog post

fchollet commented:
> I think we should keep this article as the very minimal to start playing with Keras.
> A second article to let people install TensorFlow and use the GPU will be required, but in a second article (IMO)

Fair point. But I think it would be easier to make the different sections all part of a single "manual" (which would mean releasing the two articles at the same time, and joining them either through highlighted links ("looking for the GP version? Here") or by literally putting them on the same page.

It would be a bad strategy to release info about CPU installs (which people are generally able to do without issue in 2 lines of bash) and nothing about GPU. [People don't like that](https://www.reddit.com/r/MachineLearning/comments/462p41/pros_and_cons_of_keras_vs_lasagne_for_deep/d027u0l).

PS: what do you think about using Anaconda to simplify the install? With Anaconda, which a lot of people use, the install becomes just two lines of bash. Pros/cons?
